src/web/src/features/sign-up/index.js

```diff
@@ -0,0 +1,165 @@
+import { Paper, TextField, Typography } from '@mui/material';
+import { makeStyles } from '@mui/styles';
+import { ErrorMessage, Form, Formik } from 'formik';
+import React, { useContext } from 'react';
+import * as Yup from 'yup';
+
+// workspace imports
+
+import axiosInstance from '../../configs/axios';
+import { AppContext } from '../../providers';
+
+const useStyles = makeStyles({
+  logInPaper: {
+    width: '50vw',
+    height: '120%',
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: 'center',
+    alignItems: 'center',
+    gap: '10px',
+    margin: '0 auto'
+  }
+});
+
+const SignUpSchema = Yup.object().shape({
+  email: Yup.string().email('Invalid email').required('Email required'),
+  username: Yup.string()
+    .min(4, 'Username must have at least 4 characters')
+    .required('Username required'),
+  password: Yup.string()
+    .min(6, 'Password must have at least 6 characters')
+    .required('Password required'),
+  confirmPassword: Yup.string()
+    .required('Confirm password required')
+    .oneOf([Yup.ref('password')], 'Passwords do not match')
+});
+
+const SignUp = () => {
+  const classes = useStyles();
+  const { setLoggedInUser } = useContext(AppContext);
+
+  const handleSignUp = (values, setSubmitting) => {
+    setSubmitting(false);
+    (async () => {
+      try {
+        const {
+          data: { message }
+        } = await axiosInstance.post(
+          `/user/create`,
+          {
+            username: values.username,
+            email: values.email,
+            password: values.password
+          },
+          {
+            headers: {
+              Content: 'application/json'
+            }
```

Comment:
I think we can make this config in `axiosInstance` in file `'../../configs/axios'`. You can do it here or in another PR
```suggestion
axiosInstance.defaults.headers.common['Content'] = 'application/json';
```

Reply:
Thanks for the info. I'll have a look at it